### DISCUSSION TITLE
How to change auto-update strategy

### DISCUSSION THREAD

**Discussion by Ron89** [Security-Onion-Solutions/securityonion#4960] [Unsupported Versions] [answered]
In the initial configuration, I configured it so that updates are automatically applied every 8 hours. Later I realized that it is too aggressive. But I couldn't find it on online resources on where this configuration could be changed. I also see that the auto-update is not based on mechanisms like yum-cron. Anyone know what component controls the auto updates and how to change its configuration?

**Comment by m0duspwnens** [comment, marked as answer]
@Ron89 Thanks for bringing this issue to our attention. We have added the ability to adjust this automatic schedule via the `patch` pillar to the dev branch ([PR Here](https://github.com/Security-Onion-Solutions/securityonion/pull/4990)), and it will be in the 2.3.70 release.  We will update the documentation on how to customize that prior to the release.

In the meantime, if you want to keep the schedule set to auto, but adjust the frequency at which it runs, you can modify the Salt code on the manager. Log into the manager, and as root, open `/opt/so/saltstack/default/salt/patch/os/schedule.sls` and modify [this line](https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/patch/os/schedule.sls#L45) to the frequency that you want in hours. Once that is done, you can run `salt-call state.highstate` on the nodes to update the schedule. To confirm the new schedule, you can run `salt-call schedule.list` on the nodes.

We typically wouldn't recommend this method for modifying desired functionality since the change will be overwritten the next time `soup` is run. However, the next update will overwrite this line and provide the functionality that we desire so it will be ok. Just keep in mind that once you update to 2.3.70, this schedule will need to be set in the `patch:os` pillar since the default will still be 8 hours. The documentation will be updated for how to do that closer to the release date.

**Reply by Ron89** [reply]
Thank you for the instructions. Will look forward to the documentation and SO update.
